### PR TITLE
Added project files

### PR DESCRIPTION
**Pull request by rafay-qayyum**

### CODE REVIEW

File: atm.rb [hunk 1]
```diff
@@ -0,0 +1,534 @@
+require 'csv'
+
+module Validators
+  def self.number?(num)
+    if num.match(/\d+/)
+      return true
+    else
+      return false
```
Comment (saadali7865) [inline atm.rb]: return true if num.match() else false

File: atm.rb [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    end
+  end
+  def self.get_number(mess)
+    puts mess
```
Comment (saadali7865) [inline atm.rb]: remove puts block

File: atm.rb [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+    num=gets
+    if !number?(num)
+      puts "Must be Integer.."
```
Comment (saadali7865) [inline atm.rb]: this function should only return num, printing should be handled where we are calling this method

File: atm.rb [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+class UserAccount
+  def initialize(name,atm_num,pin,expiry_date,balance)
+    @name=name
+    @atm_num=atm_num
+    @pin=pin
+    @expiry_date=expiry_date
+    @balance=balance
+  end
+  def name
+    return @name
```
Comment (saadali7865) [inline atm.rb]: no need to use return statement, simply @name will works too

**Review by jahanzaib-malik-pakwheels — CHANGES_REQUESTED** [review]
Split into different files

File: atm.rb [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+      end
+    end
+    write_data(machines)
+  end
+  def get_cash_available(id)
+    machines=read_data()
+    machines.each do |machine|
+      if machine.id==id
+        return machine.cash_available
+      end
```
Comment (saadali7865) [inline atm.rb]: return machine.cash_available if machine.id==id

File: atm.rb [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+    end
+  end
+end
+
+class AccountsController
+  def initialize()
+  end
```
Comment (saadali7865) [inline atm.rb]: empty method?

File: atm.rb [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+  def create_account(account)
+    accountsHandler=AccountsCsvDBHandler.new
+    if accountsHandler.atm_num_exists?(account.atm_num)
+      return false
+    end
```
Comment (saadali7865) [inline atm.rb]: same as above, move in single line

File: atm.rb [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+    accountsHandler.create_account(account)
+    return true
+  end
+  def update_balance(atm_num,amount)
+    accountsHandler=AccountsCsvDBHandler.new
+    account=accountsHandler.read_account(atm_num)
+    if account==nil
+      return false
+    end
```
Comment (saadali7865) [inline atm.rb]: prefer single liner

File: atm.rb [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+  end
+
+  def withdraw()
+    amount=Validators.get_number("Enter Amount to Withdraw: ")
+    if amount==nil
+      return
+    end
+    if !@atmController.withdraw(@atm_num,@machine_id,amount)
+      puts "Error Withdrawing Amount.."
```
Comment (saadali7865) [inline atm.rb]: same as above

**Review by saadali7865 — CHANGES_REQUESTED** [review]

File: atm.rb [hunk 10]
```diff
@@ -0,0 +1,534 @@
...
+  def initialize()
+    @accounts_file='accounts.csv'
+  end
+  def read_account(atm_num)
+    CSV.foreach(@accounts_file,headers:true) do |row|
+      if row['atm_num'].to_i==atm_num
+        return UserAccount.new(row['name'],row['atm_num'].to_i,row['pin'].to_i,row['expiry_date'],row['balance'].to_i)
+      end
+    end
+    return nil
```
Comment (Yasoob01) [inline atm.rb]: no need to return nil. It will return nil by default

File: atm.rb (hunk 7 above)
Comment (Yasoob01) [inline atm.rb]: you can simply do !accountsHandler.atm_num_exists?(account.atm_num)

File: atm.rb [hunk 11]
```diff
@@ -0,0 +1,534 @@
...
+    accountsHandler.delete_account(account)
+    return true
+  end
+  def login(atm_num,pin)
+    accountsHandler=AccountsCsvDBHandler.new
+    account=accountsHandler.verify_account(atm_num,pin)
+    if account==nil
+      return false
+    end
+    return true
```
Comment (Yasoob01) [inline atm.rb]: return account.present?

File: atm.rb [hunk 12]
```diff
@@ -0,0 +1,534 @@
...
+  end
+  def get_balance(atm_num)
+    accountsHandler=AccountsCsvDBHandler.new
+    account=accountsHandler.read_account(atm_num)
+    if account==nil
+      return false
+    end
+    return account.balance
```
Comment (Yasoob01) [inline atm.rb]: account.try(:balance)

File: atm.rb [hunk 13]
```diff
@@ -0,0 +1,534 @@
...
+  end
+  def get_name(atm_num)
+    accountsHandler=AccountsCsvDBHandler.new
+    account=accountsHandler.read_account(atm_num)
+    if account==nil
+      return false
+    end
+    return account.name
```
Comment (Yasoob01) [inline atm.rb]: account.try(:name)

File: atm.rb (hunk 7 above)
Comment (Yasoob01) [inline atm.rb]: instead of creating AccountCsvDBHandler in every function. Make instance variable and initilize it on object creation

**Review by Yasoob01 — CHANGES_REQUESTED** [review]

**Pull request merged**